### PR TITLE
JClassPatcher: Resize rendering related arrays, remove hacks, increase stability

### PR DESCRIPTION
**Pull request by OrN**
- End of days replays no longer flicker black periodically
- Most replay controls are more responsive
- Most replay related crashes are fixed
- Render bug from the client running too long fixed [Still testing, it takes awhile to reproduce]
- Client will no longer crash from window managers on Linux not abiding by windowing hints
- Fixes client soft lock when replay disconnects
- Exception handler for RSC client that gives us more detailed exception stacktrace (Dev)
- Bytecode tracer to determine exception handler path it took in the JVM (shows exact instructions executed, will show loop unrolled for example, and follow branches) (Dev)

### CODE REVIEW

File: src/Game/ReplayServer.java [hunk 1]
```diff
@@ -251,14 +250,8 @@ public void run() {
       client.close();
       sock.close();
       input.close();
-      if (ReplayQueue.currentIndex >= ReplayQueue.queue.size()) {
```
[the remark this answers is not in the document]
Comment (OrN) [reply src/Game/ReplayServer.java:254]: It was moved to Client.java on the rsc thread

**Comment by Hubcapp** [comment]
LGTM, going to let it sit a day for Stability:tm:

**Comment by OrN** [comment]
Same, see you tommorow.

**Comment by Hubcapp** [comment]
Everything in the OP is a True Improvement:tm: except
* Render bug from the client running too long fixed [Still testing, it takes awhile to reproduce]

Despite being the namesake of this branch, renderbug remains an issue in RSC+.

**Comment by spkaeros** [comment]
In the untouched Jagex 235 client's game engine, the main game loop which is tasked with controlling framerate and processing input events and the like (eXemplar refactor had said loop in GameShell.java, for those unfamiliar with the client) there had been some rather significant changes to the way the client would throttle framerate when it is operating at high efficiency.  The refactored 23X of RSCLegacy codebase had named the augmented sleep call shadowSleep and plopped it in as a static member of some utility class if you'd like to see the exact change I am speaking of I suggest to go look there to see what I'm talking about.  This shadowSleep routine I believe had the intent of improving efficiency or maybe improving battery life, but in any event, the result falls short of meeting target expectations; when replacing the shadowSleep call with a sane Thread.sleep, as in 204 and prior, it fixes a number of graphical problems present in the modernized revisions, such as stuttery framerate, some not-really-noticeable timing changes, and probably most important is that I believe shadowSleep causes some form of crash bug, but unfortunately can not specify any more intimate details on it.  Has there been any efforts for RSCPlus to replace the shadowSleep call with a call to Thread.sleep?  I haven't checked the code and I'm very tired at present, but I figured I'd bring it up since I was looking through your thread here.
Cheers, 
Hixk

**Pull request merged**